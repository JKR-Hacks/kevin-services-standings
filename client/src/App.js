/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import Standings from './components/Standings';
import FullStandings from './components/FullStandings';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      teams: [],
      view: 'main',
    };
    this.renderView = this.renderView.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    fetch('espn/teamstandings')
      .then(res => res.json())
      .then((data) => {
        this.setState({
          teams: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  handleClick() {
    const { view } = this.state;
    this.setState({
      view: 'fullstandings',
    });
  }

  // eslint-disable-next-line consistent-return
  renderView() {
    const { view, teams } = this.state;
    if (view === 'main') {
      return (
        <div id="main">
          <div id="main-standings">
            <Standings teams={teams} handleClick={this.handleClick} />
          </div>
        </div>
      );
    } if (view === 'fullstandings') {
      return (
        <div id="full-standings">
          <FullStandings teams={teams} />
        </div>
      );
    }
  }

  render() {
    return (
      this.renderView()
    );
  }
}
