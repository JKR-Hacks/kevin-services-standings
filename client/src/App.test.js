/* eslint-disable no-unused-vars */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import App from './App';
import Standings from './components/Standings';

describe('App component', () => {
  it('Should render App component correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should update a state in a callback', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ scores: 'new' }, () => {
      expect(wrapper.state()).toEqual({ scores: 'new', teams: [], view: 'main' });
    });
  });

  test('Should render a small label', () => {
    const wrapper = shallow(
      <App small>This is a test!</App>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call componentDidMount once', () => {
    const spy = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('Should update state view when handleClick() is called', () => {
    const wrapper = shallow(<App />);
    wrapper.instance().handleClick();
    expect(wrapper.state()).toEqual({ teams: [], view: 'fullstandings' });
  });
});
