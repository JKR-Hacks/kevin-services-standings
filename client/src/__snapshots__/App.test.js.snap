// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  id="main"
>
  <div
    id="main-standings"
  >
    <Standings
      handleClick={[Function]}
      teams={Array []}
    />
  </div>
</div>
`;
